In [1]:
import numpy as np
import pandas as pd

data = pd.read_json('data.json')
# df = pd.DataFrame(data)

# data

In [15]:
# We'll access restaurants and do a function on each object and replace it
df = pd.DataFrame(data)
ratings_array = []
def expenses(val):
    print(val)
    obj = val
    # obj = obj['restaurants']
    cost = 0
    cnt = 0
    if(type(obj) == float):
        # NaN value preprocessed
        return 0
    
    if('rating' in obj):
        lc_obj = {"name" : obj['name'], "rating": 0 , "address" : "Not in data"}
        try:
            lc_obj['rating'] = float(obj['rating'])
            cnt = cnt+1
            ratings_array.append(lc_obj)
            
        except:
            lc_obj = lc_obj
        
        try:
            lc_obj['address'] = obj['address']
        except:
            lc_obj = lc_obj
        
        return lc_obj
    else:
       return np.nan
        
    
# 12 problem 
df2 = df
df2 = df2['Patna']
# print(df2)
df2 = df2.iloc[1]
# print(df2)
df2 = pd.DataFrame(df2)
# print(df2)
df2 = df2.transpose()
df2 = df2.sort_values(by = 'rating' , ascending = False)
print(df2[:10])
# df2 = df2['restaurants'].apply(expenses)

                                                   name rating rating_count   
290590  Kwality Walls Frozen Dessert and Ice Cream Shop    4.6  1K+ ratings  \
550814                                    Slurpy Shakes    4.6  50+ ratings   
550713                    Cupcake Bliss Cake & Desserts    4.5  20+ ratings   
560369                                   GARAGE KITCHEN    4.5  50+ ratings   
553020                                     Burger It Up    4.5  20+ ratings   
561789                      Kouzina Kafe The Food Court    4.5  50+ ratings   
561782                                  Indiana Burgers    4.4  50+ ratings   
492420                       Pure Veg Meals by Lunchbox    4.4  20+ ratings   
469839                                 KINGDOM OF MOMOS    4.4  50+ ratings   
550907                                 The Dessert Zone    4.4  50+ ratings   

         cost                                            address   
290590  ₹ 100  Kwality Walls Frozen Dessert and Ice Cream Sho.

Problem 13 : Top 10 Restaurant in Patna w.r.t rating 

Kwality Walls Frozen Dessert and Ice Cream Shop    4.6  
Slurpy Shakes    4.6  
Cupcake Bliss Cake & Desserts    4.5  
GARAGE KITCHEN    4.5  
Burger It Up    4.5  
Kouzina Kafe The Food Court    4.5  
Indiana Burgers    4.4  
Pure Veg Meals by Lunchbox    4.4  
KINGDOM OF MOMOS    4.4  
The Dessert Zone    4.4  